src/index/Vocabulary.h

```diff
@@ -207,7 +207,7 @@ class Vocabulary {
 
   Id getValueIdForLE(const string& indexWord) const {
     Id lb = lower_bound(indexWord);
-    if (at(lb) != indexWord && lb > 0) {
+    if ((lb < _words.size() && at(lb) != indexWord) && lb > 0) {
```

Comment:
shouldn't the `lb > 0` come before accessing in `at(lb)` so the short circuit saves us from an illegal access?

Reply:
lb is unsigned so this should not be a problem. the greater than 0 check is only to avoid an underflow when decreasing lb later on.

Reply:
I see